gas/config/obj-elf.c

```diff
@@ -704,7 +704,7 @@ obj_elf_change_section (const char *name,
   flags = md_elf_section_flags (flags, attr, type);
 #endif
 
-  if (linkonce)
+  if (type == SHT_COMDAT || linkonce)
```

Comment:
I don't think it is a good place for this code, as you add backend specific code into general part.